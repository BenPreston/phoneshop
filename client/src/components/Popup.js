import React from 'react'

function Popup({ selected, closePopup }) {
	return (
		<section className="popup">
			<div className="content">
				<h2>{ selected.name } <span>({ selected.manufacturer })</span></h2>
				<p className="price">Price: {selected.price}</p>
				<div className="phoneDetail">
					<img src={require(`../images/${selected.imageFileName}`)} alt={`${selected.name}`} />
					<p>{selected.description}</p>
					<p><span className="spectitles">Color: </span>{selected.color}</p>
					<p><span className="spectitles">Screen: </span>{selected.screen}</p>
					<p><span className="spectitles">Processor: </span>{selected.processor}</p>
					<p><span className="spectitles">RAM: </span>{selected.ram}</p>
				</div>
				<button className="close" onClick={closePopup}>Close</button>
			</div>
		</section>
	)
}

export default Popup