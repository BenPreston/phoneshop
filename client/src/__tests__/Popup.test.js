import React from 'react'
import Popup from '../components/Popup';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';


const randomSelectedPhone = 
    {
        "id":0,
        "name":"iPhone 7",
        "manufacturer":"Apple",
        "description":"iPhone 7 dramatically improves the most important aspects of the iPhone experience. It introduces advanced new camera systems. The best performance and battery life ever in an iPhone. Immersive stereo speakers. The brightest, most colorful iPhone display. Splash and water resistance*. And it looks every bit as powerful as it is. This is iPhone 7.",
        "color":"black",
        "price":769,
        "imageFileName":"IPhone_7.png",
        "screen":"4,7 inch IPS",
        "processor":"A10 Fusion",
        "ram":2
     };

const mockClosePopup = jest.fn();

describe('<Popup />', () => {

    const wrapper = shallow(<Popup selected={randomSelectedPhone} closePopup={mockClosePopup} />);

    it('Renders and displays properly', () => {
   
        expect(toJSON(wrapper)).toMatchSnapshot();

    })

    
    it('Should have a close button that works', () => {
        const divToClick = wrapper.find('.close');

        divToClick.simulate('click');
        
        expect(mockClosePopup).toHaveBeenCalled()
      });


})