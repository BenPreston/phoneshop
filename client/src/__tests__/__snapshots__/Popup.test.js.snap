// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popup /> Renders and displays properly 1`] = `
<section
  className="popup"
>
  <div
    className="content"
  >
    <h2>
      iPhone 7
       
      <span>
        (
        Apple
        )
      </span>
    </h2>
    <p
      className="price"
    >
      Price: 
      769
    </p>
    <div
      className="phoneDetail"
    >
      <img
        alt="iPhone 7"
        src="IPhone_7.png"
      />
      <p>
        iPhone 7 dramatically improves the most important aspects of the iPhone experience. It introduces advanced new camera systems. The best performance and battery life ever in an iPhone. Immersive stereo speakers. The brightest, most colorful iPhone display. Splash and water resistance*. And it looks every bit as powerful as it is. This is iPhone 7.
      </p>
      <p>
        <span
          className="spectitles"
        >
          Color: 
        </span>
        black
      </p>
      <p>
        <span
          className="spectitles"
        >
          Screen: 
        </span>
        4,7 inch IPS
      </p>
      <p>
        <span
          className="spectitles"
        >
          Processor: 
        </span>
        A10 Fusion
      </p>
      <p>
        <span
          className="spectitles"
        >
          RAM: 
        </span>
        2
      </p>
    </div>
    <button
      className="close"
      onClick={[Function]}
    >
      Close
    </button>
  </div>
</section>
`;
